lib/handler/mruby/http_request.c

```diff
@@ -408,10 +430,31 @@ static h2o_httpclient_body_cb on_head(h2o_httpclient_t *client, const char *errs
 
     qsort(headers, num_headers, sizeof(headers[0]), headers_sort_cb);
     post_response(ctx, status, headers, num_headers, header_requires_dup);
+
     return on_body;
 }
 
-static h2o_httpclient_head_cb on_connect(h2o_httpclient_t *client, const char *errstr, h2o_iovec_t *method, h2o_url_t *url,
+static h2o_httpclient_body_cb on_head(h2o_httpclient_t *client, const char *errstr, int version, int status, h2o_iovec_t msg,
+                                      h2o_header_t *headers, size_t num_headers, int header_requires_dup)
+{
+    struct st_h2o_mruby_http_request_context_t *ctx = client->data;
+    if (can_dispose_context(ctx)) {
+        ctx->client = NULL;
+        dispose_context(ctx);
+        return NULL;
+    }
```

Comment:
I see this code repeated lots of times, wonder if we can avoid repeating the same thing.

For example, would it make sense to do something like:
```
if (!context_is_alive_or_dispose(ctx))
    return NULL;
```
while defining `context_is_alive_or_dispose` as a function that checks if the context is alive, and if not, disposes it?

Admittedly, the name sounds terse (we should preferably choose a better one), but IMO being terse is better than having the same logic copy-pasted.

Reply:
Thank you for the comment. I did it in https://github.com/h2o/h2o/pull/1975/commits/90e24bfbf53a05271e46dd1a435c632303c0db41